src/OpenLoco/Windows/Vehicle.cpp

```diff
@@ -942,6 +944,78 @@ namespace OpenLoco::Ui::Vehicle
             return self;
         }
 
+        static void cloneVehicle(window* self)
+        {
+            static loco_global<uint16_t, 0x0113642A> _113642A;
+            auto head = Common::getVehicle(self);
+            Vehicles::Vehicle train(head);
+            Vehicles::VehicleHead* newHead = nullptr;
+
+            // Get total cost for a new vehicle (note this should be removed when turned into a game command)
+            currency32_t totalCost = 0;
+            for (auto& car : train.cars)
+            {
+                auto cost = GameCommands::queryDo_5(car.front->object_id, -1);
+                if (cost == GameCommands::FAILURE)
+                {
+                    totalCost = GameCommands::FAILURE;
+                }
+                else
+                {
+                    totalCost += cost;
+                }
+            }
+
+            if (totalCost == GameCommands::FAILURE)
+            {
+                return;
+            }
+
+            // Check total cost for a new vehicle, taken from GameCommands::loc_4313C6 (note this should be removed when turned into a game command)
+            registers regs2;
+            regs2.ebp = totalCost;
+            call(0x0046DD06, regs2);
```

Comment:
The error message this presents in case there isn't enough money is "Can't sell". Can we customise this?

Reply:
Hmm. Not too sure on that. I don't want to play about with this too much as all these issues will be resolved if this becomes a full game command.